src/Symfony/Bundle/SecurityBundle/Security/FirewallConfig.php

```diff
@@ -37,11 +38,12 @@
      * @param string|null $provider
      * @param string|null $context
      * @param string|null $entryPoint
+     * @param array|null  $switchUser
      * @param string|null $accessDeniedHandler
      * @param string|null $accessDeniedUrl
      * @param string[]    $listeners
      */
-    public function __construct($name, $userChecker, $requestMatcher = null, $securityEnabled = true, $stateless = false, $provider = null, $context = null, $entryPoint = null, $accessDeniedHandler = null, $accessDeniedUrl = null, $listeners = array())
+    public function __construct($name, $userChecker, $requestMatcher = null, $securityEnabled = true, $stateless = false, $provider = null, $context = null, $entryPoint = null, $switchUser = null, $accessDeniedHandler = null, $accessDeniedUrl = null, $listeners = array())
```

Comment:
The class is final, but I guess it's still a BC break to change the arguments order if someone is instantiating this class himself somewhere.

Also, considering you don't really need the `switch_user` config, but simply know if it's enabled or not, I think you can avoid adding this and simply search for `switch_user` in the array of listeners.